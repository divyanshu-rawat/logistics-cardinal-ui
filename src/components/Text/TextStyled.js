import styled from 'styled-components';

const textSizeVariant = (size, theme) => {
  switch (size) {
    case 'xsmall':
      return theme.textFontSizeXSmall;
    case 'small':
      return theme.textFontSizeSmall;
    case 'large':
      return theme.textFontSizeLarge;
    case 'xlarge':
      return theme.textFontSizeXLarge;
    default:
      return theme.textFontSize;
  }
};

const TextStyled = styled('span')`
  color: ${({ theme, color }) => color || theme.textColor};
  font-family: ${({ theme, fontFamily }) => fontFamily || theme.textFontFamily};
  font-weight: ${({ theme, weight }) => weight || theme.textFontWeight};
  text-align: ${({ align }) => align || 'left'};
  text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'auto')};
  margin: ${({ margin, theme }) => margin || theme.textSpacing};
  font-size: ${({ fontSize, theme }) => textSizeVariant(fontSize, theme)};
  display: ${({ block }) => (block ? 'block' : 'inline-block')};

  ${({ truncate }) => {
    if (truncate) {
      return `
        display: block;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
      `;
    }

    return '';
  }};
`;

export default TextStyled;
