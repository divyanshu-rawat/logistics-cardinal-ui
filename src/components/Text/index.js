import React from 'react';
import PropTypes from 'prop-types';
import TextStyled from './TextStyled';

const Text = ({ children, ...props }) => {
  const RenderAsProp = TextStyled.withComponent(props.as);
  return <RenderAsProp {...props}>{children}</RenderAsProp>;
};

Text.propTypes = {
  children: PropTypes.any.isRequired,
  /** Final Element HTML to be rendered */
  as: PropTypes.string.isRequired,
  /** Text Color */
  color: PropTypes.string,
  /** Font Family, default is the theme one */
  fontFamily: PropTypes.string,
  /** Font Weight, default is the theme one */
  weight: PropTypes.number,
  /** Text Align, default is left */
  align: PropTypes.string,
  /** Truncate... */
  truncate: PropTypes.bool,
  /** Text Transform, UPPERCASE */
  uppercase: PropTypes.bool,
  margin: PropTypes.string,
};

Text.defaultProps = {
  children: '',
  as: 'span',
};

export default Text;
