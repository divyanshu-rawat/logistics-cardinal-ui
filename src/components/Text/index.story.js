import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import centered from '@storybook/addon-centered';
import {
  withKnobs,
  color,
  boolean,
  select,
  number,
} from '@storybook/addon-knobs';

import withTests from '../../utils/withTests';
import Theme from '../../themes/rooster';

import Text from '.';

storiesOf('Text', module)
  .addDecorator(withTests('Text/tests/index', 'TextStyled'))
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .add(
    'Simple',
    withInfo()(() => (
      <div style={{ width: '500px' }}>
        <Text
          color={color('Color', Theme.fontFamily.color)}
          fontFamily={select('Font Family', [
            Theme.fontFamily.default,
            'Times New Roman',
          ])}
          weight={number('FontWeight', Theme.fontWeight.regular)}
          align={select('Text Align', ['left', 'right', 'center'], 'left')}
          truncate={boolean('Truncate', false)}
          uppercase={boolean('Uppercase', false)}
        >
          Mussum Ipsum, cacilds vidis litro abertis. Em pé sem cair, deitado sem
          dormir, sentado sem cochilar e fazendo pose. Praesent malesuada urna
          nisi, quis volutpat erat hendrerit non. Nam vulputate dapibus.
          Detraxit consequat et quo num tendi nada. Interagi no mé, cursus quis,
          vehicula ac nisi.
        </Text>
      </div>
    )),
  );
