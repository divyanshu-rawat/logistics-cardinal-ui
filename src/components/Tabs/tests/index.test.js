import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Text from '../../Text';
import ComposedTabs from '..';

const renderedComponent = () =>
  shallow(
    <ComposedTabs
      options={{
        forecast: (
          <Text as="p">
            Lorem Ipsum is simply dummy text of the printing and typesetting
            industry.
          </Text>
        ),
        history: (
          <Text as="p">
            There are many variations of passages of Lorem Ipsum available, but
            the majority have suffered alteration in some form, by injected
            humour.
          </Text>
        ),
      }}
    />,
  );

describe('<ComposedTabs />', () => {
  it('should render the component', () => {
    const tree = toJson(renderedComponent());
    expect(tree).toMatchSnapshot();
  });

  it('`onClick` should update `state.active`', () => {
    const component = renderedComponent();
    component.instance().onClick(1)();

    expect(component.state('active')).toBe(1);
  });
});
