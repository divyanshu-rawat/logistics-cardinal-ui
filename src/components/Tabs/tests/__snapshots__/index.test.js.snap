// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComposedTabs /> should render the component 1`] = `
<HorizontalTabContainer
  active={0}
  onClick={[Function]}
  options={
    Object {
      "forecast": <Text
        as="p"
    >
        Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </Text>,
      "history": <Text
        as="p"
    >
        There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour.
    </Text>,
    }
  }
/>
`;
